### ISSUE TITLE
Add Redis address to scrapeRedisHost error message

### ISSUE THREAD

**Issue by neomantra** [oliver006/redis_exporter#638]
If a Redis host is not [available to scrape](https://github.com/oliver006/redis_exporter/blob/a1c28b775760f2f00fce07a24db7fd4e83c26b9f/exporter/exporter.go#L530), this error is logged:
```
Couldn't connect to redis instance
```

Looking through issues here, it seems this is often because of `--redis.addr=` and the suggestion is to enable `--debug` to check.   However, that generates a *ton* of output (which is great when one needs it). 

 In my case, we had a misconfigured static target and simply having the host in the message would have helped.

Is there any reason (security?) the message cannot be like:
```
log.Errorf("Couldn't connect to redis instance %s", e.redisAddr)
```

**What version of redis_exporter are you running?**
Please run `redis_exporter --version` if you're not sure what version you're running.
[ ] 0.3x.x
[X] 1.x.x
docker image v1.24.0

Thanks!

**Comment by oliver006** [comment]
Reason is that you can put the password in the URI and logging the full URI would leak the password to the logs.

An improvement could be to log the URI with the password redacted. Happy to review/merge a PR for that but I'm currently short of time to implement something myself.

**Comment by neomantra** [comment]
That makes sense -- I didn't want to proceed with a PR without understanding it...  and now I made the PR :)

**Comment by oliver006** [comment]
Closed by #643
